llvm/lib/Target/AIE/AIE2RegisterInfo.cpp

```diff
@@ -126,6 +126,21 @@ bool AIE2RegisterInfo::isSimplifiableReservedReg(MCRegister PhysReg) const {
                               AIE2::mSRmRegClass.contains(PhysReg));
 }
 
+bool AIE2RegisterInfo::isReservedStickyReg(MCRegister PhysReg) const {
+  switch (PhysReg) {
+  case AIE2::srCompr_uf:
+  case AIE2::srSparse_of:
+  case AIE2::srF2FFlags:
+  case AIE2::srF2IFlags:
+  case AIE2::srFPFlags:
+  case AIE2::srSRS_of:
+  case AIE2::srUPS_of:
+    return true;
+  default:
+    return false;
+  }
+}
```

Comment:
Could this just be `return AIE2::mSRmRegClass.contains(PhysReg)`?

Reply:
Yes if we can include srCarry. Semantically we can use, because we detect register reads, however If I remember correctly, this represents another unexpected freedom behavior.